src/sagemaker/jumpstart/utils.py

```diff
@@ -1525,3 +1535,63 @@ def wrapped_f(*args, **kwargs):
     if _func is None:
         return wrapper_cache
     return wrapper_cache(_func)
+
```

Comment:
Please add unit tests for both methods in this module, there is a lot of conditional logic to verify

Reply:
yep will do